### ISSUE TITLE
Complex numbers with norm, abs and abs2

### ISSUE THREAD

**Issue by OlivierHnt** [JuliaIntervals/IntervalArithmetic.jl#245]
I reported an issue on Julia discourse yesterday, but I feel like I should also mentioned it here. The link:

https://discourse.julialang.org/t/complex-number-abs-and-intervalarithmetic/18268?u=olivierhnt

Despite abs and norm giving odd output, abs2 seems to work fine.
Am I missing something ?

[Copied from Discourse:

```
julia> abs.([0..0.01] + [0..1]*im)
1-element Array{Interval{Float64},1}:
 [0, ∞]

julia> norm([0..0.01] + [0..1]*im, 1)
 [0, ∞]
```

However,  there is no issue if I use `abs2()`:

```
julia> abs2.([0..0.01] + [0..1]*im)
1-element Array{Interval{Float64},1}:
 [0, 1.00011]
```

I did not see anything on the Issue section of IntervalArithmetic.jl but perhaps I am missing something ?
Should `norm` be avoided when using IntervalArithmetic ?

**Comment by Kolaru** [comment]
It is because Julia compute the norm of `a + b im` as essentially

```jl
r = a/b
b^2 * sqrt(1 + r^2)
```

and explicitely check if `iszero(b)`, which is always false for interval (the same problem keep coming back).

However `sqrt(abs2(a + b im))` works fine (it is not tight however as `abs2(z) = real(z)*real(z) + imag(z)*imag(z)`, but this is less urgent to fix).

**Comment by dpsanders** [comment]
Fixed by #250.

Note that you were using arrays and this is unnecessary. With the branch from #250, I get:

```jl
julia> x = (0..0.01) + (0..1)*im
[0, 0.0100001] + [0, 1]*im

julia> abs(x)
[0, 1.00005]

julia> norm(x)
[0, 1.00005]
```

**Comment by dpsanders** [comment]
Fixed by #250
